lib/generateNormals.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+var findAccessorMinMax = require('./findAccessorMinMax');
+var mergeBuffers = require('./mergeBuffers');
+var readAccessor = require('./readAccessor');
+var uninterleaveAndPackBuffers = require('./uninterleaveAndPackBuffers');
+var Cesium = require('cesium');
+var Cartesian3 = Cesium.Cartesian3;
+var defined = Cesium.defined;
+var WebGLConstants = Cesium.WebGLConstants;
+module.exports = generateNormals;
+
+var scratchNormal = new Cartesian3();
+
+// Generate normals if they do not exist.
+function generateNormals(gltf) {
+    var i;
+    var j;
+    var generatedNormals = false;
+    var primitives = getPrimitives(gltf);
+    var primitivesLength = primitives.length;
+    for (i = 0; i < primitivesLength; ++i) {
+        var primitive = primitives[i];
+        var positionAccessorId = primitive.attributes.POSITION;
+        var normalAccessorId = primitive.attributes.NORMAL;
+        var indicesAccessorId = primitive.indices;
+        if (primitive.mode === WebGLConstants.TRIANGLES && defined(indicesAccessorId) && defined(positionAccessorId) && !defined(normalAccessorId)) {
+            generatedNormals = true;
+            var indicesAccessor = gltf.accessors[indicesAccessorId];
+            var positionAccessor = gltf.accessors[positionAccessorId];
+            var indices = readAccessor(gltf, indicesAccessor).data;
+            var positions = readAccessor(gltf, positionAccessor).data;
+
+            // Generate one normal for each position
+            var normalsLength = positions.length;
+            var trianglesLength = indices.length / 3;
+
+            // Initialize the normals to 0
+            var normals = new Array(normalsLength);
+            for (j = 0; j < normalsLength; j++) {
+                normals[j] = new Cartesian3();
+            }
+
+            // Add face normal to each vertex normal
+            for (j = 0; j < trianglesLength; ++j) {
+                var index1 = indices[j * 3 + 0];
+                var index2 = indices[j * 3 + 1];
+                var index3 = indices[j * 3 + 2];
+                var faceNormal = getFaceNormal(positions[index1], positions[index2], positions[index3], scratchNormal);
```

Comment:
Maybe I'm misreading, but does this just set the vertex normal to be the face normal? That won't be the correct vector.